ejb/src/test/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJBTest.java

```diff
@@ -0,0 +1,108 @@
+import biz.karms.sinkit.ioc.*;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.sort.SortBuilder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Tomas Kozel
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ArchiveServiceEJBTest {
+
+    //TODO: Shouldn't log messages be also shown (could be useful for debugging)
+    @Mock
+    private Logger log;
+
+    @Mock
+    private ElasticService elasticService;
+
+    @Mock
+    private LogstashClient logstashClient;
+
+    @InjectMocks
+    private ArchiveServiceEJB archiveService;
+
+
+/**
+ * Blacklist tests
+ */
+    @Test
+    public void deactivateRecordTest() throws Exception{
+
+        IoCRecord ioc= getIoCRecord(null, "facebook.malware.com","SomeFeed",
+                1, "malware");
+        archiveService.deactivateRecord(ioc);
+        assertEquals(false,ioc.isActive());
+        // important for documentId to be recomputed
+        assertEquals(IoCIdentificationUtils.computeHashedId(ioc),
+                ioc.getDocumentId());
+        verify(elasticService).delete(ioc,
+                archiveService.ELASTIC_IOC_INDEX,
+                archiveService.ELASTIC_IOC_TYPE);
+        verify(elasticService).index(ioc,
+                archiveService.ELASTIC_IOC_INDEX,
+                archiveService.ELASTIC_IOC_TYPE);
+        verifyNoMoreInteractions(elasticService);
+    }
+
+    /**
+     * creates IoCRecord observed ageHours in the past
+     * @param ip
+     * @param fqdn
+     * @param sourceName feed name
+     * @param ageHours sets time.observation to current
+     * @return
+     */
+    public IoCRecord getIoCRecord(String ip, String fqdn, String sourceName, int ageHours, String classificationType) {
+        IoCRecord ioc = new IoCRecord();
+
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.HOUR,-ageHours);
+        Date observation = cal.getTime();
+        IoCTime time = new IoCTime();
+        ioc.setTime(time);
+        ioc.getTime().setObservation(observation);
+
+        IoCClassification classification = new IoCClassification();
+        classification.setType(classificationType);
+        ioc.setClassification(classification);
+
+        ioc.setSource(new IoCSource());
+        ioc.getSource().setIp(ip);
+        ioc.getSource().setFQDN(fqdn);
+        ioc.setFeed(new IoCFeed());
+        ioc.getFeed().setName(sourceName);
+        return ioc;
+    }
+
+    @Test
+    public void test() {
+        System.out.println(System.getProperty("java.io.tmpdir"));
```

Comment:
@hugeox What does this method test?

Reply:
This is something that I just copied from Tomas Kozel, and it makes some sense to me since it tests whether everything is rightly configured. If this test fails, we know that there is some problem not connected with the right functioning of our code.

Reply:
@hugeox Remove those, please.